### PR TITLE
[hail][feature] add `contig_length` function

### LINKED ISSUE #7111: how do I get a contig's end position?

### PR DESCRIPTION
**Pull request by tpoterba**
Fixes #7111

### CODE REVIEW

File: hail/src/main/scala/is/hail/expr/ir/functions/ReferenceGenomeFunctions.scala [hunk 1]
```diff
@@ -85,5 +85,11 @@ class ReferenceGenomeFunctions(rg: ReferenceGenome) extends RegistryFunctions {
           Seq(TString(), TInt32())).get
         If(isValid(Array(contig, pos)), getRef(Array(contig, pos, before, after)), NA(TString()))
     }
+
+    registerRGCode("contigLength", TString(), TInt32(), null) {
```
Comment (danking) [inline hail/src/main/scala/is/hail/expr/ir/functions/ReferenceGenomeFunctions.scala]: what's the deal with this `pt` argument which is `null` and not used in the definition of `registerRGCode`?

File: hail/src/main/scala/is/hail/expr/ir/functions/ReferenceGenomeFunctions.scala [hunk 2]
```diff
@@ -85,5 +85,11 @@ class ReferenceGenomeFunctions(rg: ReferenceGenome) extends RegistryFunctions {
...
+      case (r, rt, (contigT, contig: Code[long])) =>
```
Comment (danking) [inline hail/src/main/scala/is/hail/expr/ir/functions/ReferenceGenomeFunctions.scala]: what's up with the lowercase long?

File: hail/src/main/scala/is/hail/expr/ir/functions/ReferenceGenomeFunctions.scala (hunk 1 above)
Comment (tpoterba) [reply hail/src/main/scala/is/hail/expr/ir/functions/ReferenceGenomeFunctions.scala]: this is the ptype registration function. If `null`, the function returns the canoncical PType for the virtual type.

File: hail/src/main/scala/is/hail/expr/ir/functions/ReferenceGenomeFunctions.scala (hunk 2 above)
Comment (tpoterba) [reply hail/src/main/scala/is/hail/expr/ir/functions/ReferenceGenomeFunctions.scala]: a bug is up

Comment (tpoterba) [reply hail/src/main/scala/is/hail/expr/ir/functions/ReferenceGenomeFunctions.scala]: hah, was being interpreted as a type variable.

**Pull request merged**